src/dune_rules/compilation_context.ml

```diff
@@ -114,10 +115,31 @@ let bin_annot t = t.bin_annot
 
 let context t = Super_context.context t.super_context
 
+type rename =
+  { new_name : Module_name.t
+  ; toplevel_modules : Module_name.t list
```

Comment:
I don't anticipate that anybody will need it any time soon, and simplifying the code to work with a simpler data structure seems worth it to me. It will probably make it more clear that the downstream code is reasonable.